src/main.rs

```diff
@@ -100,61 +130,48 @@ fn main() -> ! {
                 format!("{:.3}", user_value).blue()
             );
 
-            let found_var = vars.iter_mut().find(|x| x.repr == user_repr);
-            if let Some(found_var) = found_var {
-                // Reassign
-                found_var.value = user_value;
-            } else {
-                // Assign
-                let user_var = Variable {
-                    repr: user_repr,
-                    value: user_value,
-                };
-                vars.push(user_var);
-            }
-
-            continue;
+            assign_var(vars, user_value, user_repr);
+            
+            Ok(assign_confirmation)
         }
-
-        let (x, repr) = match doeval(&input, &vars) {
-            Ok((a, b)) => (a, b),
-            Err(e) => {
-                handle_errors(e, input, 0);
-                continue;
-            }
-        };
+    else {
+        // Evaluate as normal
+        let result = doeval(&input, &vars);
+        if let Err(Error::Parsing(idx)) = result {
+            return Err(Error::Parsing(idx));
+        }
+        let (x, repr) = result?;
 
         let formatted = stringify(&repr, color_cli);
+        let eval_string = format!("[ {} ] => {}", formatted, format!("{:.3}", x).blue());
 
         vars[0].value = x; // Set ans to new value
 
-        println!("[ {} ] => {}", formatted, format!("{:.3}", x).blue());
+        Ok(eval_string)
     }
 }
 
-fn handle_errors(e: Error, input: String, offset: usize) {
-    // Offset is added so that highlighting can be added to expressions that come after an '=' during assignment
+fn handle_errors(e: Error, input: String) {
```

Comment:
```suggestion
fn handle_errors(error: Error, input: &str) {
```

This function should probably take a `&str` instead, and as a personal preference I'd like to see a longer-than-one variable name here.

Reply:
Fixed with 16a1aaaa8cf7bb41804a6107c0c580ce5d3004b6